Exiled.API/Features/Toys/Light.cs

```diff
@@ -0,0 +1,119 @@
+// -----------------------------------------------------------------------
+// <copyright file="Light.cs" company="Exiled Team">
+// Copyright (c) Exiled Team. All rights reserved.
+// Licensed under the CC BY-SA 3.0 license.
+// </copyright>
+// -----------------------------------------------------------------------
+
+namespace Exiled.API.Features.Toys
+{
+    using AdminToys;
+    using Mirror;
+    using UnityEngine;
+
+    /// <summary>
+    /// A wrapper class for light source Admin Toys.
+    /// </summary>
+    public class Light
+    {
+        /// <summary>
+        /// Initializes a new instance of the <see cref="Light"/> class.
+        /// </summary>
+        public Light()
+        {
+            Base = Object.Instantiate(ToysHelper.LightBaseObject);
+        }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="Light"/> class from a <see cref="LightSourceToy"/>.
+        /// </summary>
+        /// <param name="toy">The toy to be wrapped.</param>
+        public Light(LightSourceToy toy)
+        {
+            Base = toy;
+        }
+
+        /// <summary>
+        /// Gets the base <see cref="LightSourceToy"/>.
+        /// </summary>
+        public LightSourceToy Base { get; }
+
+        /// <summary>
+        /// Gets or sets the position of the light.
+        /// </summary>
+        public Vector3 Position
+        {
+            get => Base.NetworkPosition;
+            set
+            {
+                Base.transform.position = value;
+                Base.NetworkPosition = value;
```

Comment:
Dont use network stuff, use transform, the network stuff is automatically synced, therefore if you change it directly via network... it wont change its transform so youre desyncing it from the server some frames until the updateserver method is called and its network things are synced with its transform